Build without assembly optimization fails on x64 due to AESCE being enabled

### Summary

Currently the no assembly build of Cryptofuzz on OSS-Fuzz is failing. It complains about AESCE being enabled, while this is a x64 system. Shouldn't AESCE be disabled by default on x64?

The reproduction steps below still used to work a while ago.

### System information

x64 Ubuntu with Clang

### Expected behavior

No build failure.

### Actual behavior

Build failure:

```
/home/jhg/cf-mbed/mbedtls/include/mbedtls/check_config.h:74:2: error: #error "MBEDTLS_AESCE_C defined, but not all prerequisites"
   74 | #error "MBEDTLS_AESCE_C defined, but not all prerequisites"
```

### Steps to reproduce

```sh
git clone --depth 1 -b development https://github.com/ARMmbed/mbedtls.git
cd mbedtls/
scripts/config.pl set MBEDTLS_PLATFORM_MEMORY
scripts/config.pl set MBEDTLS_CMAC_C
scripts/config.pl set MBEDTLS_NIST_KW_C
scripts/config.pl set MBEDTLS_ARIA_C
scripts/config.pl unset MBEDTLS_HAVE_ASM
scripts/config.pl unset MBEDTLS_PADLOCK_C
scripts/config.pl unset MBEDTLS_AESNI_C
mkdir build/
cd build/
cmake .. -DENABLE_PROGRAMS=0 -DENABLE_TESTING=0
make -j$(nproc)
```

### Additional information

--- a/include/mbedtls/check_config.h
+++ b/include/mbedtls/check_config.h
@@ -70,8 +70,11 @@
 #error "MBEDTLS_AESNI_C defined, but not all prerequisites"
 #endif
 
+#if defined(__aarch64__) && defined(__GNUC__)
+/* We don't do anything with MBEDTLS_AESCE_C on systems without ^ these two */
 #if defined(MBEDTLS_AESCE_C) && !defined(MBEDTLS_HAVE_ASM)
 #error "MBEDTLS_AESCE_C defined, but not all prerequisites"
+#endif
 #endif
 
 #if defined(MBEDTLS_CTR_DRBG_C) && !defined(MBEDTLS_AES_C)

